fineract-provider/src/main/java/org/apache/fineract/portfolio/shareaccounts/domain/ShareAccount.java

```diff
@@ -224,7 +224,7 @@ public boolean setSubmittedDate(final Date submittedDate) {
 
     public boolean setApprovedDate(final Date approvedDate) {
         boolean toReturn = false;
-        if (!this.approvedDate.equals(approvedDate)) {
+        if (this.approvedDate.compareTo(approvedDate) == 0 ? Boolean.FALSE : Boolean.TRUE) {
             this.approvedDate = approvedDate;
```

Comment:
very very minor suggestion, can we not keep it just !=0 (this makes it little confusing IMO).

Reply:
Yeah, but I think this way is clearer.